lib/org/chromium/webidl/Canonicalizer.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'Canonicalizer',
+
+  documentation: `This component expects to receive AST nodes from various
+    sources. Canonical versions of interface definitions will be created
+    from AST nodes of the same source.
+    Please note that this step cannot be distributed. All computations
+    must be done through one canonicalizer instances, or you will
+    get fragmented results.`,
+
+  requires: [
+    'org.chromium.webidl.ast.Enum',
+    'org.chromium.webidl.ast.Typedef',
+    'org.chromium.webidl.PipelineMessage',
+  ],
+
+  properties: [
+    {
+      class: 'Function',
+      documentation: `Function is called after waitTime seconds has passed
+        since the last call to addFragment.`,
+      name: 'onDone',
+      required: true,
+    },
+    {
+      class: 'Int',
+      documentation: `The amount of time to wait in seconds after the last
+        addFragment request, before onDone() is called with the canonical
+        IDL files.`,
+      name: 'waitTime',
+    },
+    {
+      documentation: `A map of maps containing canonical IDL files. The
```

Comment:
I don't know if this is a good structure. It feels like you could have one instance of this for each source instead of having one instance that holds a map of things for each source.